canonical-webteam/generator-canonical-webteam
# Unify run script

While thinking about #32, adding Flask support, it occurred to me that the way I've separate out the parts of the run script in the generator project is probably doing more harm than good:

- Each project type uses a specific generator (e.g. `run-django-db`)
- Each project type ends up with a customer version of the run script

It occurred to me that this makes the run script template complex to understand and edit, as it's a mixture of Bash and [EJS](http://ejs.co/) syntax, and it also makes it harder to add the run script to a project, and that run scripts can't be naiively copied between projects. And the benefits in reducing the amount of redundant code aren't that great - most projects end up sharing the majority of the run script code anyway.

So I've decided to unify the run script into one script that will work for all project types - instead of having a different script per project type, it will dynamically detect what type of project it's running in at run time - e.g. by checking for a `package.json` file.

QA
--

To use this version of the `canonical-webteam` module, clone this branch, and then in this branch's folder, run:

```
sudo npm link
```

Now pull down a few different project types that currently work with the run script, and try adding this run script instead with:

```
yo canonical-webteam:run
```

Now try out all the run functions and check they work as expected (e.g. start with `./run help`).

Afterwards, reset your `canonical-webteam` module by going back into this branch's folder and running:

```
sudo npm unlink
```
